### ISSUE TITLE
Switch to markdown syntax for literate programming

### ISSUE THREAD

**Issue by denismerigoux** [CatalaLang/catala#64]
Right now, the literate programming syntax of Catala uses the following operators:
- `@@` to open and close headings (with `+` for precedence)
- `@`  to open and close article titles
- `/*` and `*/` to delimit code sections

This syntax is non-standard and does not display well anywhere. Since we have  to pick a syntax, might as well pick one that enhances interoperability with existing tools.

## Towards a markdown syntax

Markdown seems adapted to the level of formatting that we want for the legislative part. Plus, it's supported by most IDEs and has special GitHub/Gitlab support so it would help the display on the CVS.

With a markdown syntax, a Catala file could look like:

````md
## Some law header

### Some law header subsection

#### [Article 6 ] 

```
declaration structure:
   ...
```
````

The `[...]` notation is used to distinguish law headers from titles of legislative article. An article is the atom of legislation, all the code blocks should belong to an article (see #29)

## How to implement that

We need to change the parts of the lexers that deal with this syntax like 

https://github.com/CatalaLang/catala/blob/7a835ac5f654e42beb82bf25f570606136584c54/src/catala/catala_surface/lexer.ml#L539-L585

From the top of my head no changes to the parser should be required. EDIT: `BEGIN_CODE` and `END_CODE` should be merged together I think ? Or we can say that `BEGIN_CODE` should always be ` ```catala ` whereas `END_CODE` is ` ``` `

**Comment by EmileRolley** [comment]
I think this issue can be closed now.

**Comment by denismerigoux** [comment]
Closed by #84
